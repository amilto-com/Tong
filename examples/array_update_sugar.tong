// Placeholder example for forthcoming in-place style array element update sugar.
// Currently TONG does not support direct arr[i] = v syntax. This example shows the
// manual pattern of rebinding for a single-element update.
// Planned sugar: arr[i] = expr  ==>  let tmp = arr; let arr = update_index(tmp, i, expr)
// or a dedicated runtime intrinsic returning a new array with the substituted element.

fn main() {
  var xs = [0,1,2]
  xs[0] = xs[0] + 10
  xs[2] = xs[1] + xs[2]
  print("updated array", xs[0], xs[1], xs[2])
}

main()
