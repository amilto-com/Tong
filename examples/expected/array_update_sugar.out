updated array 10 1 3